apps/interop-rxjs-signal/src/app/list/photos.component.ts

```diff
@@ -80,40 +91,88 @@ import { PhotoStore } from './photos.store';
       </section>
     </ng-container>
   `,
-  providers: [provideComponentStore(PhotoStore)],
   host: {
     class: 'p-5 block',
   },
 })
-export default class PhotosComponent implements OnInit {
-  store = inject(PhotoStore);
-  readonly vm$ = this.store.vm$.pipe(
-    tap(({ search }) => {
-      if (!this.formInit) {
-        this.search.setValue(search);
-        this.formInit = true;
-      }
-    })
+export default class PhotosComponent extends SignalState<PhotoComponentState> {
+  private readonly photoService = inject(PhotoService);
+  protected readonly vm = this.selectMany(
+    ['photos', 'search', 'page', 'pages', 'loading', 'error'],
+    ({ photos, search, page, pages, loading, error }) => {
+      return {
+        photos,
+        search,
+        pages,
+        page,
+        loading,
+        error,
+        endOfPage: page === pages,
+      };
+    }
   );
+  protected readonly searchCtrl = new FormControl();
 
-  private formInit = false;
-  search = new FormControl();
-
-  ngOnInit(): void {
-    this.store.search(
-      this.search.valueChanges.pipe(
-        skipWhile(() => !this.formInit),
-        debounceTime(300),
-        distinctUntilChanged()
+  constructor() {
+    super();
```

Comment:
you can get the localStorage here and initialize your state depending on the value. 

And effect is perfect for saving into the localstorage when search or page update, isn't it ?

Reply:
That makes total sense